Panic on unknown preset in DAC template

## Issue detail
When an invalid preset is specified in a DAC YAML template, the awsdac tool hangs indefinitely instead of showing a warning and continuing processing. This occurs because the code attempts to access properties of a nil definition object after logging the "Unknown preset" warning.

## Reproduce process

### command / error messages
```
$ awsdac minimal-bad-preset.yaml -o output.png
time="2025-08-19T08:00:00Z" level=warning msg="Unknown preset NonExistentPreset on AWS::ElasticLoadBalancingV2::LoadBalancer\n"
panic: runtime error: invalid memory address or nil pointer dereference                                                                                                                                                                       
[signal SIGSEGV: segmentation violation code=0x2 addr=0x20 pc=0x100976e30]                                                                                                                                                                    
                                                                                                                                                                                                                                              
goroutine 1 [running]:                                                                                                                                                                                                                        
github.com/awslabs/diagram-as-code/internal/ctl.loadResources(0x14000126740, {0x14000035958?}, 0x14000035b08)                                                                                                                                 
        /Users/kitayu/GitHub/diagram-as-code/branches/awsdac-mcp-server/internal/ctl/create.go:273 +0xc20                                                                                                                                     
github.com/awslabs/diagram-as-code/internal/ctl.CreateDiagramFromDacFile({0x16f7cb466, 0x17}, 0x14000116500, 0x14000035bd8)                                                                                                                   
        /Users/kitayu/GitHub/diagram-as-code/branches/awsdac-mcp-server/internal/ctl/dacfile.go:137 +0x78c                                                                                                                                    
main.main.func2(0x140001cc300?, {0x140001ce240?, 0x7?, 0x1009f032d?})                                                                                                                                                                         
        /Users/kitayu/GitHub/diagram-as-code/branches/awsdac-mcp-server/cmd/awsdac/main.go:70 +0x118                                                                                                                                          
github.com/spf13/cobra.(*Command).execute(0x140001d2008, {0x14000126050, 0x3, 0x3})                                                                                                                                                           
        /Users/kitayu/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:983 +0x830                                                                                                                                                          
github.com/spf13/cobra.(*Command).ExecuteC(0x140001d2008)                                                                                                                                                                                     
        /Users/kitayu/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x344                                                                                                                                                         
github.com/spf13/cobra.(*Command).Execute(...)                                                                                                                                                                                                
        /Users/kitayu/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039                                                                                                                                                                
main.main()                                                                                                                                                                                                                                   
        /Users/kitayu/GitHub/diagram-as-code/branches/awsdac-mcp-server/cmd/awsdac/main.go:86 +0x34c    
```

### Yaml template
```yaml
Diagram:
  DefinitionFiles:
    - Type: URL
      Url: "https://raw.githubusercontent.com/awslabs/diagram-as-code/main/definitions/definition-for-aws-icons-light.yaml"

  Resources:
    Canvas:
      Type: AWS::Diagram::Canvas
      Children:
        - TestResource
    
    TestResource:
      Type: AWS::ElasticLoadBalancingV2::LoadBalancer
      Preset: NonExistentPreset
      Title: "Test Resource"
```

Expected behavior: Tool should show warning about unknown preset and continue processing to generate the diagram.

Actual behavior: Tool hangs after showing the warning message and never completes.

Root cause: In internal/ctl/create.go around line 271, when `ds.Definitions[v.Preset] returns ok == false`, the code logs a warning but continues to access def.Fill, def.Border, etc. on a nil object, causing the hang.

--- a/internal/ctl/cfntemplate.go
+++ b/internal/ctl/cfntemplate.go
@@ -81,7 +81,7 @@ func CreateDiagramFromCFnTemplate(inputfile string, outputfile *string, generate
 	}
 
 	var ds definition.DefinitionStructure
-	var resources map[string]*types.Resource = make(map[string]*types.Resource)
+	resources := make(map[string]*types.Resource)
 
 	log.Info("--- Load DefinitionFiles section ---")
 	if opts.OverrideDefFile != "" {
@@ -92,14 +92,14 @@ func CreateDiagramFromCFnTemplate(inputfile string, outputfile *string, generate
 				Type: "URL",
 				Url:  opts.OverrideDefFile,
 			}
-			overrideDefTemplate.Diagram.DefinitionFiles = append(overrideDefTemplate.Diagram.DefinitionFiles, defFile)
+			overrideDefTemplate.DefinitionFiles = append(overrideDefTemplate.DefinitionFiles, defFile)
 		} else {
 			log.Infof("As given overrideDefFile, use %s as LocalFile instead of %v", opts.OverrideDefFile, &template.DefinitionFiles)
 			var defFile = DefinitionFile{
 				Type:      "LocalFile",
 				LocalFile: opts.OverrideDefFile,
 			}
-			overrideDefTemplate.Diagram.DefinitionFiles = append(overrideDefTemplate.Diagram.DefinitionFiles, defFile)
+			overrideDefTemplate.DefinitionFiles = append(overrideDefTemplate.DefinitionFiles, defFile)
 		}
 		if err := loadDefinitionFiles(&overrideDefTemplate, &ds); err != nil {
 			return fmt.Errorf("failed to load override definition files: %w", err)
@@ -147,8 +147,8 @@ func convertTemplate(cfn_template cft.Template, template *TemplateStruct, ds def
 			resource := res.(map[string]interface{})
 			typeValue, _ := resource["Type"].(string)
 
-			if _, ok := template.Diagram.Resources[logicalId]; !ok {
-				template.Diagram.Resources[logicalId] = Resource{
+			if _, ok := template.Resources[logicalId]; !ok {
+				template.Resources[logicalId] = Resource{
 					Type: typeValue,
 				}
 			}
@@ -180,7 +180,7 @@ func convertTemplate(cfn_template cft.Template, template *TemplateStruct, ds def
 				}
 
 				//related_resource_type can not have children resources due to the restrict of definition file.
-				if !def.CFn.HasChildren {
+				if def == nil || !def.CFn.HasChildren {
 					log.Infof("%s cannot have children resource.", related)
 					continue
 				}
@@ -188,23 +188,23 @@ func convertTemplate(cfn_template cft.Template, template *TemplateStruct, ds def
 				//Find parent
 				findParent = true
 				parent_logicalId := related
-				parent_resources := template.Diagram.Resources[parent_logicalId]
+				parent_resources := template.Resources[parent_logicalId]
 				parent_resources.Children = append(parent_resources.Children, logicalId)
-				template.Diagram.Resources[parent_logicalId] = parent_resources
+				template.Resources[parent_logicalId] = parent_resources
 			}
 
 			//If there is no parent resource, consider "AWSCloud" as the parent
 			if !findParent {
-				parents := template.Diagram.Resources["AWSCloud"]
+				parents := template.Resources["AWSCloud"]
 				parents.Children = append(parents.Children, logicalId)
-				template.Diagram.Resources["AWSCloud"] = parents
+				template.Resources["AWSCloud"] = parents
 			}
 		}
 	}
 }
 
 func ensureSingleParent(template *TemplateStruct) {
-	for logicalId, resource := range template.Diagram.Resources {
+	for logicalId, resource := range template.Resources {
 
 		if logicalId == "Canvas" || logicalId == "AWSCloud" {
 			continue
@@ -213,7 +213,7 @@ func ensureSingleParent(template *TemplateStruct) {
 		if len(resource.Children) > 1 {
 
 			for _, childID := range resource.Children {
-				child, ok := template.Diagram.Resources[childID]
+				child, ok := template.Resources[childID]
 				if !ok {
 					continue
 				}
@@ -234,9 +234,9 @@ func ensureSingleParent(template *TemplateStruct) {
 						}
 					}
 
-					grandparent_resources := template.Diagram.Resources[logicalId]
+					grandparent_resources := template.Resources[logicalId]
 					grandparent_resources.Children = newChildren
-					template.Diagram.Resources[logicalId] = grandparent_resources
+					template.Resources[logicalId] = grandparent_resources
 					resource.Children = newChildren
 					log.Infof("Updated resource %s children: %v", logicalId, newChildren)
 				}
@@ -256,7 +256,7 @@ func associateCFnChildren(template *TemplateStruct, ds definition.DefinitionStru
 			continue
 		}
 
-		if !def.CFn.HasChildren {
+		if def == nil || !def.CFn.HasChildren {
 			log.Infof("%s cannot have children resource.", logicalId)
 			continue
 		}
@@ -281,16 +281,16 @@ func associateCFnChildren(template *TemplateStruct, ds definition.DefinitionStru
 				continue
 			}
 
-			if def.Border == nil {
+			if def == nil || def.Border == nil {
 				resources[logicalId].SetBorderColor(color.RGBA{0, 0, 0, 255})
 				resources[logicalId].SetFillColor(color.RGBA{0, 0, 0, 0})
 			}
 
 			// Update yaml template for providing
 			newChildren = append(newChildren, child)
-			template_resource := template.Diagram.Resources[logicalId]
+			template_resource := template.Resources[logicalId]
 			template_resource.Children = newChildren
-			template.Diagram.Resources[logicalId] = template_resource
+			template.Resources[logicalId] = template_resource
 
 		}
 	}

--- a/internal/ctl/create.go
+++ b/internal/ctl/create.go
@@ -139,7 +139,7 @@ func createDiagram(resources map[string]*types.Resource, outputfile *string) err
 func loadDefinitionFiles(template *TemplateStruct, ds *definition.DefinitionStructure) error {
 
 	// Load definition files
-	for _, v := range template.Diagram.DefinitionFiles {
+	for _, v := range template.DefinitionFiles {
 		switch v.Type {
 		case "URL":
 			log.Infof("Fetch definition file from URL: %s\n", v.Url)
@@ -191,22 +191,27 @@ func loadResources(template *TemplateStruct, ds definition.DefinitionStructure, 
 			def, ok := ds.Definitions[v.Type]
 			if !ok {
 				newType := fallbackToServiceIcon(v.Type)
-				_, check := ds.Definitions[newType]
-				if !check {
+				fallbackDef, check := ds.Definitions[newType]
+				if !check || fallbackDef == nil {
 					log.Warnf("Type %s is not defined in the DAC definition file. It cannot be fall backed to service icon. Ignore this type.\n", v.Type)
 					continue
 				}
 				log.Warnf("Type %s is not defined in the DAC definition file. It's fall backed to its service icon (Type %s).\n", v.Type, newType)
-				def = ds.Definitions[newType]
+				def = fallbackDef
 
 				// Change the title to indicate the original resource type for fallback icons.
 				if v.Title == "" {
 					resources[k].SetLabel(&v.Type, nil, nil)
 				}
 			}
-			if def.Type == "Resource" {
+			if def == nil {
+				log.Warnf("Definition for %s is nil. Skip this resource.\n", v.Type)
+				continue
+			}
+			switch def.Type {
+			case "Resource":
 				resources[k] = new(types.Resource).Init()
-			} else if def.Type == "Group" {
+			case "Group":
 				resources[k] = new(types.Resource).Init()
 			}
 			if fill := def.Fill; fill != nil {
@@ -269,52 +274,53 @@ func loadResources(template *TemplateStruct, ds definition.DefinitionStructure, 
 			def, ok := ds.Definitions[v.Preset]
 			if !ok {
 				log.Warnf("Unknown preset %s on %s\n", v.Preset, v.Type)
-			}
-			if fill := def.Fill; fill != nil {
-				fillColor, err := stringToColor(fill.Color)
-				if err != nil {
-					return fmt.Errorf("failed to parse fill color for resource %s: %w", k, err)
-				}
-				resources[k].SetFillColor(fillColor)
-			}
-			if border := def.Border; border != nil {
-				borderColor, err := stringToColor(border.Color)
-				if err != nil {
-					return fmt.Errorf("failed to parse border color for resource %s: %w", k, err)
-				}
-				resources[k].SetBorderColor(borderColor)
-				switch border.Type {
-				case "straight":
-					resources[k].SetBorderType(types.BORDER_TYPE_STRAIGHT)
-				case "dashed":
-					resources[k].SetBorderType(types.BORDER_TYPE_DASHED)
-				default:
-					resources[k].SetBorderType(types.BORDER_TYPE_STRAIGHT)
-				}
-			}
-			if label := def.Label; label != nil {
-				if label.Title != "" {
-					resources[k].SetLabel(&label.Title, nil, nil)
-				}
-				if label.Color != "" {
-					c, err := stringToColor(label.Color)
+			} else {
+				if fill := def.Fill; fill != nil {
+					fillColor, err := stringToColor(fill.Color)
 					if err != nil {
-						return fmt.Errorf("failed to parse label color for resource %s: %w", k, err)
+						return fmt.Errorf("failed to parse fill color for resource %s: %w", k, err)
 					}
-					resources[k].SetLabel(nil, &c, nil)
+					resources[k].SetFillColor(fillColor)
 				}
-				if label.Font != "" {
-					resources[k].SetLabel(nil, nil, &label.Font)
-				}
-			}
-			if headerAlign := def.HeaderAlign; headerAlign != "" {
-				resources[k].SetHeaderAlign(headerAlign)
-			}
-			if icon := def.Icon; icon != nil {
-				if def.CacheFilePath != "" {
-					err := resources[k].LoadIcon(def.CacheFilePath)
+				if border := def.Border; border != nil {
+					borderColor, err := stringToColor(border.Color)
 					if err != nil {
-						return fmt.Errorf("failed to load icon from cache file path: %w", err)
+						return fmt.Errorf("failed to parse border color for resource %s: %w", k, err)
+					}
+					resources[k].SetBorderColor(borderColor)
+					switch border.Type {
+					case "straight":
+						resources[k].SetBorderType(types.BORDER_TYPE_STRAIGHT)
+					case "dashed":
+						resources[k].SetBorderType(types.BORDER_TYPE_DASHED)
+					default:
+						resources[k].SetBorderType(types.BORDER_TYPE_STRAIGHT)
+					}
+				}
+				if label := def.Label; label != nil {
+					if label.Title != "" {
+						resources[k].SetLabel(&label.Title, nil, nil)
+					}
+					if label.Color != "" {
+						c, err := stringToColor(label.Color)
+						if err != nil {
+							return fmt.Errorf("failed to parse label color for resource %s: %w", k, err)
+						}
+						resources[k].SetLabel(nil, &c, nil)
+					}
+					if label.Font != "" {
+						resources[k].SetLabel(nil, nil, &label.Font)
+					}
+				}
+				if headerAlign := def.HeaderAlign; headerAlign != "" {
+					resources[k].SetHeaderAlign(headerAlign)
+				}
+				if icon := def.Icon; icon != nil {
+					if def.CacheFilePath != "" {
+						err := resources[k].LoadIcon(def.CacheFilePath)
+						if err != nil {
+							return fmt.Errorf("failed to load icon from cache file path: %w", err)
+						}
 					}
 				}
 			}

--- a/internal/ctl/dacfile.go
+++ b/internal/ctl/dacfile.go
@@ -35,12 +35,15 @@ func getTemplate(inputfile string) ([]byte, error) {
 		}()
 
 		data, err = io.ReadAll(resp.Body)
+		if err != nil {
+			return nil, err
+		}
 	} else {
 		// Local file
 		data, err = os.ReadFile(inputfile)
-	}
-	if err != nil {
-		return nil, err
+		if err != nil {
+			return nil, err
+		}
 	}
 	return data, nil
 }
@@ -103,7 +106,7 @@ func CreateDiagramFromDacFile(inputfile string, outputfile *string, opts *Create
 	}
 
 	var ds definition.DefinitionStructure
-	var resources map[string]*types.Resource = make(map[string]*types.Resource)
+	resources := make(map[string]*types.Resource)
 
 	log.Info("Load DefinitionFiles section")
 	if opts.OverrideDefFile != "" {
@@ -114,14 +117,14 @@ func CreateDiagramFromDacFile(inputfile string, outputfile *string, opts *Create
 				Type: "URL",
 				Url:  opts.OverrideDefFile,
 			}
-			overrideDefTemplate.Diagram.DefinitionFiles = append(overrideDefTemplate.Diagram.DefinitionFiles, defFile)
+			overrideDefTemplate.DefinitionFiles = append(overrideDefTemplate.DefinitionFiles, defFile)
 		} else {
 			log.Infof("As given overrideDefFile, use %s as LocalFile instead of %v", opts.OverrideDefFile, &template.DefinitionFiles)
 			var defFile = DefinitionFile{
 				Type:      "LocalFile",
 				LocalFile: opts.OverrideDefFile,
 			}
-			overrideDefTemplate.Diagram.DefinitionFiles = append(overrideDefTemplate.Diagram.DefinitionFiles, defFile)
+			overrideDefTemplate.DefinitionFiles = append(overrideDefTemplate.DefinitionFiles, defFile)
 		}
 		if err := loadDefinitionFiles(&overrideDefTemplate, &ds); err != nil {
 			return fmt.Errorf("failed to load override definition files: %w", err)

--- a/internal/definition/definition_structure.go
+++ b/internal/definition/definition_structure.go
@@ -47,8 +47,10 @@ func (ds *DefinitionStructure) LoadDefinitions(filePath string) error {
 			return ""
 		}()
 		if src != "" {
-			sourceDefinition := b.Definitions[src]
-			v.Parent = sourceDefinition
+			sourceDefinition, exists := b.Definitions[src]
+			if exists {
+				v.Parent = sourceDefinition
+			}
 		}
 		b.Definitions[k] = v
 	}

